nexus_lifecycle_manager/include/nexus_lifecycle_manager/lifecycle_manager.hpp

```diff
@@ -117,11 +116,14 @@ class Implementation
     {
       std::optional<uint8_t> state = client->get_state();
       if (!state.has_value())
+      {
         ok = false;
+        continue;
```

Comment:
can we log a warning telling users that we were unable to get the state for this client?
Second should we add this client to a list and then remove these clients outside this loop?

Reply:
https://github.com/osrf/nexus/pull/52/commits/967fade404187ce19724bd929b71588bcdf89cb6.
WRT the removing not sure, I noticed that there is an API to remove nodes, it's just that it's not used anywhere in the codebase https://github.com/osrf/nexus/blob/c3200d5f2abdd57339699e461b41b552a9d0e163/nexus_lifecycle_manager/include/nexus_lifecycle_manager/lifecycle_manager.hpp#L512-L516

I'd like to try to introduce it first to do proper cleanup (i.e. with unregistration) rather than "silently" cleaning up automatically. Nexus tends to have _a lot_ of output in the terminal and I fear a single one off warning during bringup might be missed otherwise, and lead to more difficult to debug issued down the line

Reply:
Fair enough.